mirdata/annotations.py

```diff
@@ -319,38 +544,239 @@ def validate_lengths_equal(array_list):
             raise ValueError("Arrays have unequal length")
 
 
-def validate_confidence(confidence):
+def validate_tempos(tempo, tempo_unit):
+    """Validate if tempos are well-formed
+
+    Args:
+        tempo (list): list of tempo values
+        tempo_unit (str): tempo unit, one of TEMPO_UNITS
+
+    Raises:
+        ValueError: if tempos are not well-formed
+    """
+    validate_unit(tempo_unit, TEMPO_UNITS)
+    if (tempo < 0).any():
+        raise ValueError("tempos must be positive")
+
+
+def validate_beat_positions(positions, position_unit):
+    """Validate if positions is well-formed.
+
+    Args:
+        positions (np.ndarray): an array of positions values
+        positions_unit (str): one of BEAT_POSITION_UNITS
+
+    Raises:
+        ValueError: if positions values are incompatible with the unit
+
+    """
+    validate_unit(position_unit, BEAT_POSITION_UNITS)
+
+    position_shape = np.shape(positions)
+    if len(position_shape) != 1:
+        raise ValueError(
+            f"positions should be 1d, but array has shape {position_shape}"
+        )
+
+    if (positions < 0).any():
+        raise ValueError("beat positions must be positive. Found values below 0.")
+
+    if position_unit in ["bar_index", "global_index"] and not np.array_equal(
+        np.floor(positions), positions
+    ):
+        raise ValueError(
+            "measure index or global indexes should be integers. "
+            + "Found fractional values."
+        )
+
+    if position_unit == "bar_index" and np.max(positions) > 16:
```

Comment:
Why 16?

Reply:
I was thinking 16 was the max common number of beats per bar. Maybe I'm wrong? :)

Reply:
I think this works for the datasets we have, but if later one some `MTG`'s Indian music datasets are added we might see some issues (IIRC those can have >20 beats in a cycle). We can change this number then  :)

Reply:
Ah ok! I'll update it now! it was mostly to catch the difference between bar_index and global_index.